improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -574,34 +550,88 @@ def test_complex_condition(self):
+                            lwe_sleetfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                thresholds[0],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                        "-",
+                        iris.Constraint(
+                            name="probability_of_rainfall_rate_above_threshold",
+                            rainfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                thresholds[1],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                    ],
+                    ">=",
+                    0.5,
+                ],
+                [
+                    [
+                        iris.Constraint(
+                            name="probability_of_lwe_sleetfall_rate_above_threshold",
+                            lwe_sleetfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                thresholds[2],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                        "-",
+                        iris.Constraint(
+                            name="probability_of_lwe_snowfall_rate_above_threshold",
+                            lwe_snowfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                thresholds[3],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                    ],
+                    ">=",
+                    0.5,
+                ],
+            ],
+            "OR",
+        ]
         self.assertIsInstance(result, list)
-        self.assertTrue(all([isinstance(s, str) for s in result]))
-        self.assertEqual(len(result), 1)
-        self.assertEqual(result[0], expected)
-
-
-class Test_construct_extract_constraint(IrisTest):
+        self.assertEqual(len(result), 2)
+        self.assertIsInstance(result[0], list)
+        for i in range(2):
+            for j in range(2):
+                for k in [0, 2]:
+                    constraint_exp = expected[0][i][0][k]
+                    constraint_res = result[0][i][0][k]
+                    self.assertArrayEqual(
+                        self.cubes.extract(constraint_res)[0].data,
+                        self.cubes.extract(constraint_exp)[0].data,
+                    )
+                self.assertEqual(result[0][i][0][1], expected[0][i][0][1])
+            self.assertEqual(result[0][j][1:], expected[0][j][1:])
```

Comment:
It seems that the loop variable from line 621 is being used outside the loop. Is this intended? Maybe just set it to 1, if that's the case.

Reply:
Yep, you're right